addons/stock/stock.py

```diff
@@ -895,8 +895,10 @@ def test_assigned(self, cr, uid, ids):
                 if all([x.state != 'waiting' for x in pick.move_lines]):
                     return True
             for move in pick.move_lines:
-                if (move.state) == 'waiting':
-                    move.check_assign()
```

Comment:
There is a way to set the context on a browse record but it is undocumented for all I know. I don't remember what it is but I suspect that this method sets the context on all records which is not what we want. And it's just syntactic sugar, right?

Maybe we can wait for @hbrunn's review to enlighten us.